File: howManyGames_hacker_rank.rb
def howManyGames(p, d, m, s)
    if p > s
        return 0
    else
        count = 1
        s -= p 
    end
    while s > 0
        if p - d < m
            p = m
        else 
            p -=d
        end

        if s - p < 0
            break
        else
            s -= p
            count += 1
        end
    end
    count
end

File: howManyGames_hacker_rank.js
function howManyGames(p, d, m, s) {

    s -= p;
    let count = 1;
    while (s > 0) {
        if (p - d >= m) {
            p -= d;
        } else {
            p = m;
        }
        if (s - p >= 0) {
            count += 1
            s -= p
        } else {
            break
        }

    }
    return count;
}